ports/mimxrt/machine_sdcard.c

```diff
@@ -0,0 +1,225 @@
+    //  with PFD0_clk = PLL2_clk * 18 / N
+    //       PFD0_clk = 528MHz   * 18 / 24 = 396MHz
+    CLOCK_InitSysPfd(kCLOCK_Pfd0, 24U);
+
+    #if defined MICROPY_USDHC1 && USDHC1_AVAIL
+    // Configure USDHC clock source and divider
+    CLOCK_SetDiv(kCLOCK_Usdhc1Div, 1U);  // USDHC_input_clk = PFD0_clk / 2 = 198MHZ
+    CLOCK_SetMux(kCLOCK_Usdhc1Mux, 1U);  // Select PFD0 as clock input for USDHC
+    #endif
+
+    #if defined MICROPY_USDHC2 && USDHC2_AVAIL
+    // Configure USDHC clock source and divider
+    CLOCK_SetDiv(kCLOCK_Usdhc2Div, 1U);  // USDHC_input_clk = PFD0_clk / 2 = 198MHZ
+    CLOCK_SetMux(kCLOCK_Usdhc2Mux, 1U);  // Select PFD0 as clock input for USDHC
+    #endif
+
+    // Initialize USDHC
+    const usdhc_config_t config = {
+        .endianMode = kUSDHC_EndianModeLittle,
+        .dataTimeout = 0xFU,
+        .readWatermarkLevel = 128U,
+        .writeWatermarkLevel = 128U,
+    };
+    USDHC_Init(self->usdhc_inst, &config);
+
+    sdcard_init(self, 198000000UL);  // Initialize SDCard Host with 198MHz base clock
+    sdcard_init_pins(self);
+
+    self->initialized = false;
+    return sdcard_detect(self);
+}
+
+STATIC mp_obj_t sdcard_obj_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *all_args) {
+    mp_map_t kw_args;
+    mp_map_init_fixed_table(&kw_args, n_kw, all_args + n_args);
+
+    // Parse args
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args, all_args, &kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    // Extract arguments
+    mp_int_t sdcard_id = args[SDCARD_INIT_ARG_ID].u_int;
+
+    if (!(1 <= sdcard_id && sdcard_id <= MP_ARRAY_SIZE(mimxrt_sdcard_objs))) {
+        nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_ValueError, "SDCard(%d) doesn't exist", sdcard_id));
+    }
+
+    mimxrt_sdcard_obj_t *self = &mimxrt_sdcard_objs[(sdcard_id - 1)];
+
+    // Initialize SDCard Host
+    if (machine_sdcard_init_helper(self, args)) {
+        return MP_OBJ_FROM_PTR(self);
+    } else {
+        return mp_const_none;
+    }
+}
+
+// init()
+STATIC mp_obj_t machine_sdcard_init(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    // Parse args
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args) - 1];
+    mp_arg_parse_all(n_args, pos_args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    // Todo: make sure that there is a valid reason that a call to init would not change anything for an already initialized hsot
+    machine_sdcard_init_helper(pos_args[0], args);
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_KW(machine_sdcard_init_obj, 1, machine_sdcard_init);
+
+// deinit()
+STATIC mp_obj_t machine_sdcard_deinit(mp_obj_t self_in) {
+    // TODO: Implement deinit function
+    return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(machine_sdcard_deinit_obj, machine_sdcard_deinit);
+
+// readblocks(block_num, buf, [offset])
```

Comment:
this is not the signature as implemented below, the `offset` arg is not supported